File: backend/app/schemas/region.py
from datetime import datetime

from pydantic import BaseModel

from backend.app.schemas.geometry import GeoJSONGeometry


class Region(BaseModel):
    id: str
    name: str
    state: str | None = None
    country: str | None = None
    geometry: GeoJSONGeometry
    created_at: datetime


class RegionsResponse(BaseModel):
    regions: list[Region]

File: frontend/src/types/region.ts
import { GeoJSONGeometry } from './geojson';

export interface Region {
  id: string;
  name: string;
  state?: string | null;
  country?: string | null;
  geometry: GeoJSONGeometry;
  created_at?: string;
}

export interface RegionsResponse {
  regions: Region[];
}
